packages/injector/src/injector.ts

```diff
@@ -808,7 +837,8 @@ export class Injector implements InjectorInterface {
                     }
                     current = current.indexAccessOrigin.container;
                 }
-                return () => config;
+
+                if (config) return () => config;
```

Comment:
Had to change this to support creating factories for interfaces without having to `Inject<.>` every property. Interface properties always have a `property.type.indexAccessOrigin` set, which triggers config lookup. Without this change it would always return undefined (which was undesirable behavior in the first place I think)